flink-ml-lib/src/main/java/org/apache/flink/ml/params/knn/KnnClassifierParams.java

```diff
@@ -0,0 +1,4 @@
+package org.apache.flink.ml.params.knn;
+
+/** knn train parameters. */
+public interface KnnClassifierParams<T> extends KnnTrainParams<T>, KnnPredictParams<T> {}
```

Comment:
1. XXXX Params is the params of algorithm, not infra

2. predict is the action of model.

3. Estimator is the api algo user can see. train is the inner process in the Estimator.

4. Classifier is one kind of estimator, Regressor and Cluster also can be an estimator.

Reply:
Model's action is currently expressed by the API `Model::transform`. It is probably simpler to avoid introducing multiple names for the same thing (e.g. predict/transform).

And if we believe there is semantic difference between classifier, regressor and estimator, which I believe makes sense, it is probably useful to explicitly add abstract classes and express the semantics difference via API and Java doc, as is the case in Spark.

Reply:
1. predict is a inner concept, not a api (like transform)
2. I think: estimator is the abstract classes of  classifier, regressor